engine/engine/content/builtins/scripts/edn.lua

```diff
@@ -146,90 +146,98 @@ local function encode_as_primitive(val)
   return encoder(val, val_type, mt)
 end
 
-local function encode_edn_table(val, key)
+local function encode_edn_table(ref, ref_table)
   local encoded_kvs = {}
-  for k, v in pairs(val) do
+  for k, v in pairs(ref_table) do
     encoded_kvs[#encoded_kvs + 1] = encode_as_primitive(k)
     encoded_kvs[#encoded_kvs + 1] = encode_as_primitive(v)
   end
-  local encoded_str = encode_string(str(key, type(key), debug.getmetatable(key)))
+  local encoded_str = encode_string(str(ref, type(ref), debug.getmetatable(ref)))
   return "#lua/table{:content [" .. table.concat(encoded_kvs, " ") .. "] :string " .. encoded_str .. "}"
 end
 
-local function encode_refs(refs, keys)
+local function encode_refs(ref_to_table)
   local encoded_refs = {}
-  local val
-  for i = 1, #refs do
-    val = refs[i]
-    encoded_refs[#encoded_refs + 1] = encode_as_primitive(val)
-    encoded_refs[#encoded_refs + 1] = encode_edn_table(keys[val], val)
+  for ref, ref_table in pairs(ref_to_table) do
```

Comment:
As I understand, now  `ref_to_table` is fairly small, so we use `pairs` again?

Reply:
Yes, it's that, plus `ref_to_table` is now a "public" data structure, i.e. it's used in `mobdebug.lua`, so I wanted to get some semantical simplicity over optimization.

Reply:
👍